src/socket.rs

```diff
@@ -100,6 +108,16 @@ pub struct Open;
 pub struct Connected;
 
 impl<A: NetworkAddress, S> Socket<A, S> {
+    pub fn local_addr(&self) -> std::io::Result<A> {
+        let addr = self.socket.get_ref().getsockname()?;
+        A::from_sockaddr(addr, PrivateToken).ok_or_else(|| std::io::ErrorKind::Other.into())
+    }
+
+    pub fn peer_addr(&self) -> std::io::Result<A> {
+        let addr = self.socket.get_ref().getpeername()?;
+        A::from_sockaddr(addr, PrivateToken).ok_or_else(|| std::io::ErrorKind::Other.into())
+    }
+
```

Comment:
is this error effectively unreachable? otherwise maybe give a proper message. it would be hard to debug otherwise if we ever hit this.

Reply:
should be effectively unreachable, replicates behaviour rust has in the standard library (gives a similar error on similar conditions).